wallet/test/Test/Pos/Util/MnemonicSpec.hs

```diff
@@ -65,31 +79,38 @@ spec = do
         it "Can generate 128 bits entropy" $
             (length . entropyToByteString <$> genEntropy @128) `shouldReturn` 16
 
-        it "Mnemonic to JSON" $ forM_ testVectors $ \(bytes, _, mnemonic, _, _) ->
+        it "Mnemonic to JSON" $ forM_ testVectors $ \(bytes, _, _, mnemonic, _, _) ->
             Aeson.encode mnemonic `shouldBe` bytes
 
-        it "Mnemonic from JSON" $ forM_ testVectors $ \(bytes, _, mnemonic, _, _) ->
+        it "Mnemonic from JSON" $ forM_ testVectors $ \(bytes, _, _, mnemonic, _, _) ->
             Aeson.decode bytes `shouldBe` pure mnemonic
 
-        it "CBackupPhrase to JSON" $ forM_ testVectors $ \(bytes, _, mnemonic, _, _) ->
+        it "CBackupPhrase to JSON" $ forM_ testVectors $ \(bytes, _, _, mnemonic, _, _) ->
             Aeson.encode (CBackupPhrase mnemonic) `shouldBe` jsonV0Compat bytes
 
-        it "CBackupPhrase from JSON" $ forM_ testVectors $ \(bytes, _, mnemonic, _, _) ->
+        it "CBackupPhrase from JSON" $ forM_ testVectors $ \(bytes, _, _, mnemonic, _, _) ->
             Aeson.decode (jsonV0Compat bytes) `shouldBe` pure (CBackupPhrase mnemonic)
 
-        it "Mnemonic to Entropy" $ forM_ testVectors $ \(_, entropy, mnemonic, _, _) ->
+        it "CBackupPhrase from legacy JSON" $ forM_ testVectors $ \(_, legacyBytes, _, mnemonic, _, _) ->
+            Aeson.decode (jsonV0Compat legacyBytes) `shouldBe` pure (CBackupPhrase mnemonic)
+
+        it "Mnemonic from legacy JSON" $ forM_ testVectors $ \(_, legacyBytes, _, _, _, _) ->
+            Aeson.decode legacyBytes `shouldSatisfy` isNothing @ (Mnemonic 12)
```

Comment:
again, I am probably missunderstanding it - but shouldn't it be `Aeson.decode (jsonV0Compat legacyBytes)...`

Reply:
@akegalj It should be correct, but `it` description lacking, my bad.

So, examples:
```haskell
bytes = ["word", "word", ...]
legacyBytes = ["word word word", "word", "word", ... ] -- could look like bytes too
(jsonV0Compat legacyBytes) = {"bpToList": ["word word word", "word",  "word", ...]}
-- bytes is the format for Mnemonic
-- (jsonV0Compat legacyBytes) is the format for CBackupPhrase
```

This test makes sure that the `Mnemonic` decoding doesn't work with legacyBytes.

I could 
- Describe these concepts in a comment at the top of the file, near `TestVector`
- Change the `it` description

Reply:
Aha it makes more sense now. Maybe having an example as you showed in the example above would be nice to have in comments. Nothing too serious though